shared/analytics_tracking/base.py

```diff
@@ -9,8 +11,5 @@ class BaseAnalyticsTool(ABC):
     def is_enabled(cls):
         raise NotImplementedError()
 
-    def track_event(self, user_id, event_name, *, is_enterprise, event_data={}):
-        raise NotImplementedError()
-
-    def track_user(self, user_id, user_data={}, is_enterprise=False):
+    def track_event(self, event: Event, *, is_enterprise, context: None):
```

Comment:
(as an end user) Do I need to create the event myself before calling the `track_event`?

i.e. can I do `analytics_tool.track_event(Events.USER_SIGNED_UP, event_data={...}, is_enterprise=False)` ?

Reply:
The idea here is that the end user will have to deal with the analytics manager, and not know anything about events. The manager will take the data and wrap it in an event object and send it to the analytic tools individually. 
We probably need to rename analytics_tool to analytics_manager because that's what it really is

Reply:
Good, good. I just wanted to make sure the user didn't have to know about both the manager and the events. Thanks.